vhdl_lang/src/analysis/declarative.rs

```diff
@@ -606,7 +617,14 @@ impl<'a, 't> AnalyzeContext<'a, 't> {
                     scope.add(ent, diagnostics);
                 }
             }
+            Declaration::PackageDeclaration(ref mut unit) => {
+                self.analyze_package(unit, diagnostics)?;
+            }
+            Declaration::PackageBody(ref mut unit) => {
+                self.analyze_package_body(unit, diagnostics)?;
+            }
             Declaration::Configuration(..) => {}
+            Declaration::Disconnection(..) => {} // @TODO
```

Comment:
Probably it's out of scope for this PR to start analysing Disconnect and guarded statements thoroughly. But some limited analysis could still be done. For example, for the statement
```
disconnect foo, bar : resolved_bit after 5 ns ;
                    ^              ^                             ^
                    |                |                               Check that this is an expression of type time
                    |                Check that this is a valid type that matches the type declared by the single names
                   Check that this is a valid signal name
```

For all of this, examples should exist somewhere in the code on how to analyze this.
- For the 'expect that something is an expression of type time' part, you can have a look at [how `wait` statements are analysed](https://github.com/VHDL-LS/rust_hdl/blob/805ab71a5ce8ce7af30557dabc20728b0205c68a/vhdl_lang/src/analysis/sequential.rs#L159)
- To check the type and the signal names, you could look at [how `alias` declarations are analysed](https://github.com/VHDL-LS/rust_hdl/blob/805ab71a5ce8ce7af30557dabc20728b0205c68a/vhdl_lang/src/analysis/declarative.rs#L243), though this is not as straightforward as the time expression.

Reply:
I did the type checking and also the expression checking and also added some tests for that. Pretty cool that it works just like that. Anyway I have no clue how to check for the names of the identifier tbh.

Reply:
Done, see https://github.com/VHDL-LS/rust_hdl/pull/307#discussion_r1625150504